javascript-custom-methods/calculator.js

```diff
@@ -0,0 +1,43 @@
+/* exported calculator */
+var calculator = {
+  add: function (x, y) {
+    return x + y;
+  },
+  subtract: function (x, y) {
+    return x - y;
+  },
+  multiply: function (x, y) {
+    return x * y;
+  },
+  divide: function (x, y) {
+    return x / y;
+  },
+  square: function (x) {
+    return x * x;
+  },
+  sumAll: function (numbers) {
+    var sum;
+    for (var i = 0; i < numbers.length; i++) {
+      if (sum === undefined) {
```

Comment:
Why not initially assign `sum` a value that is numeric so that you don't need this conditional?